iron-dropdown.html

```diff
@@ -340,6 +340,49 @@
           } else {
             Polymer.IronOverlayBehaviorImpl._applyFocus.apply(this, arguments);
           }
+        },
+
+        /**
+         * An element related to the dropdown (the triggering element, or an
+         * element within the dropdown) has lost focus.
+         */
+        _activeElementBlurred: function(event) {
+          if (event.target._blurListener) {
+            // Stop listening to blur events on the target.
+            event.target.removeEventListener('blur', event.target._blurListener);
+            event.target._blurListener = null;
+          }
+          this._dropdownFocusChanged(false);
+        },
+
+        /**
+         * Return true if the focus is anywhere within the dropdown.
+         */
+        _focusWithinDropdown: function() {
+          if (this.shadowRoot) {
+            return this.shadowRoot.activeElement !== null;
+          } else {
+            var activeElement = this._manager.deepActiveElement;
+            return Polymer.dom(this).deepContains(activeElement);
+          }
+        },
+
+        /**
+         * Overridden from `IronOverlayBehavior`.
+         */
+        _onCaptureFocus: function(event) {
+          if (!this._focusWithinDropdown()) {
+            // The focus has left the dropdown, e.g., in response to the user
+            // tabbing away.
+
+            // Avoid restoring focus after is closed.
+            this._restoreFocusNode = null;
```

Comment:
From what I can see, neither clicking outside to cancel nor pressing Esc invoke `_onCaptureFocus`. Should we expect to see that method to be invoked in those conditions?